synapse/rest/client/v1/register.py

```diff
@@ -304,6 +312,51 @@ def _do_app_service(self, request, register_json, session):
             "home_server": self.hs.hostname,
         })
 
+    @defer.inlineCallbacks
+    def _do_shared_secret(self, request, register_json, session):
+        yield run_on_reactor()
+
+        if "mac" not in register_json:
```

Comment:
Given ``register_json`` is just a ``dict`` from JSON, users may not be setting the right types of these keys. We should be type checking here, which we can actually do in one fell swoop with checking for the existence of keys by doing ``isinstance(register_json.get("mac"), basestring)``